tests/unit/test_pushpull.py

```diff
@@ -93,6 +103,8 @@ def test_git_push_fail(mock_subproc_popen):
 def test_git_push_success(mock_subproc_popen):
     # Given
     process_mock = Mock()
+    env = os.environ.copy()
```

Comment:
Only commend I have is that if we could enhance these tests:

We could mock out `os.environ` to return a dummy value and then assert that subprocess was called with the dummy value + `GIT_TERMINAL_PROMPT` as the env parameter

Reply:
good catch, updated.